addons/l10n_it_edi/models/account_edi_format.py

```diff
@@ -228,41 +228,55 @@ def _l10n_it_goods_in_italy(self, invoice):
+            'TD19': dict(move_types=['in_invoice', 'in_refund'], import_type='in_invoice', simplified=False, self_invoice=True, services_or_goods="consu", goods_in_italy=True),
+            'TD28': dict(move_types=['in_invoice', 'in_refund'], import_type='in_invoice', simplified=False, self_invoice=True, partner_country_code="SM")
         }
 
-    def _l10n_it_get_document_type(self, invoice):
-        is_simplified = self._l10n_it_edi_is_simplified(invoice)
-        is_self_invoice = self._l10n_it_edi_is_self_invoice(invoice)
+    def _l10n_it_get_invoice_features_for_document_type_selection(self, invoice):
+        """ Returns a dictionary of features to be compared with the TDxx FatturaPA
+            document type requirements. """
         services_or_goods = self._l10n_it_edi_services_or_goods(invoice)
-        goods_in_italy = services_or_goods == 'consu' and self._l10n_it_goods_in_italy(invoice)
-        partner_in_eu = self._l10n_it_edi_partner_in_eu(invoice.commercial_partner_id)
-        for code, infos in self._l10n_it_document_type_mapping().items():
-            info_services_or_goods = infos.get('services_or_goods', "both")
-            info_partner_in_eu = infos.get('partner_in_eu', False)
-            if all([
-                invoice.move_type in infos.get('move_types', False),
-                # Only check downpayment if the key is specified in the document_type_mapping entry
-                # If it's not specified, the get() will return None and the condition will be True
-                infos.get('downpayment') in (None, invoice._is_downpayment()),
-                is_self_invoice == infos.get('self_invoice', False),
-                is_simplified == infos.get('simplified', False),
-                info_services_or_goods in ("both", services_or_goods),
-                info_partner_in_eu in (False, partner_in_eu),
-                goods_in_italy == infos.get('goods_in_italy', False),
-            ]):
-                return code
+        return {
+            'move_types': invoice.move_type,
+            'partner_in_eu': self._l10n_it_edi_partner_in_eu(invoice.commercial_partner_id),
+            'partner_country_code': invoice.commercial_partner_id.country_id.code,
+            'simplified': self._l10n_it_edi_is_simplified(invoice),
+            'self_invoice': self._l10n_it_edi_is_self_invoice(invoice),
+            'downpayment': invoice._is_downpayment(),
+            'services_or_goods': services_or_goods,
+            'goods_in_italy': services_or_goods == 'consu' and self._l10n_it_goods_in_italy(invoice),
+            # An invoice is deferred if its Transport Document (DDT) date comes after the invoice date.
+            # To handle those, please nstall l10n_it_stock_ddt
+            'direct_invoice': True,
+        }
 
-        return None
+    def _l10n_it_get_document_type(self, invoice):
```

Comment:
This function compares the features of the invoice to those of the Document Type